src/components/MovieList/MovieList.jsx

```diff
@@ -0,0 +1,111 @@
+import { Button, CardContainer } from 'components/MovieList/styles';
+import React, { useState } from 'react';
+
+import MovieCard from 'components/MovieCard/MovieCard';
+import PropTypes from 'prop-types';
+import blurImage from 'assets/image.jpg';
+import dayjs from 'dayjs';
+import { fetchDataFromApiList } from 'utils/function';
+
+// eslint-disable-next-line import/no-mutable-exports
+export let pageNumber = 1;
+/**
+ * MovieList Component
+ * Displays a list of movies in a paginated manner and provides an option to load more movies.
+ *
+ * @param {Array} movies - The array of movies to display.
+ * @param {Function} setMovies - Function to set the movies in the parent component.
+ * @param {Function} setParams - Function to set parameters in the parent component.
+ * @param {Object} params - Parameters for the movie list.
+ * @returns {JSX.Element} JSX element representing the movie list.
+ */
+export default function MovieList({ movies, setMovies, setParams, params }) {
+  const [isMenuOpen, setIsMenuOpen] = useState('');
```

Comment:
When I first saw the `isMenuOpen` state I was expecting its value to be boolean instead of the selected movie id. I would suggest renaming it to something like movieMenu where its value could be an object containing the status of the menu (open/closed) and the selectedMovieId. 
```

const [movieMenu, setMovieMenu] = useState({ status: 'closed', movieId: null });
const menuOpenHandler = (movieId) => {
  setMovieMenu((prevMenu) => ({
    status: prevMenu.movieId === movieId && prevMenu.status === 'open' ? 'closed' : 'open',
    movieId: movieId,
  }));
};
```

Reply:
Can I change the name of state and keep the functionality as I did?